lib/dry/types/options.rb

```diff
@@ -26,6 +26,11 @@ def with(new_options)
       def meta(data = nil)
         data ? with(meta: @meta.merge(data)) : @meta
       end
+
+      # @return [Boolean]
+      def optional?
```

Comment:
this should be split in two pieces, one is in `definition.rb` (look at [these methods](https://github.com/dry-rb/dry-types/blob/278b03588b1f1a9ccecf422f9c886c80a7f61bc5/lib/dry/types/definition.rb#L44-L52))
another is in `sum.rb`, this way we can avoid the `self.class == Sum` check